packages/flutter_driver/lib/src/driver/timeline_summary.dart

```diff
@@ -458,4 +505,73 @@ class TimelineSummary {
   RasterCacheSummarizer _rasterCacheSummarizer() => RasterCacheSummarizer(_extractNamedEvents(kRasterCacheEvent));
 
   GCSummarizer _gcSummarizer() => GCSummarizer.fromEvents(_extractEventsWithNames(kGCRootEvents));
+
+  List<Duration> _extractOverallDurations() {
+    final List<_DurationTuple> buildDurationTuples =
+        _extractEventPairs(kBuildFrameEventName, _DurationTuple.from);
+    final List<_DurationTuple> rasterDurationTuples =
+        _extractEventPairs(kRasterizeFrameEventName, _DurationTuple.from);
+    buildDurationTuples.checkSortedByFirst();
+    rasterDurationTuples.checkSortedByFirst();
+
+    final List<_DurationTuple> pipelineItemTuples =
+        _extractNamedEvents(_kPipelineItemFrameEventName)
+            .groupListsBy((TimelineEvent e) => e.json['id'] as String)
+            .values
+            .map((List<TimelineEvent> events) => _DurationTuple.maybeFrom(
+                events.firstWhereOrNull((TimelineEvent e) => e.phase == 't'),
+                events.firstWhereOrNull((TimelineEvent e) => e.phase == 'f')))
+            .whereNotNull()
+            .sortedBy((_DurationTuple tuple) => tuple.first);
```

Comment:
I vaguely understand what this is supposed to be doing, but I think it'd be much easier to read if I could acutally see the implementations.

I'm also not clear on when nulls are allowed or not here, and why we're tolerating nulls.

Reply:
IIRC I am seeing null situations in real world scenarios but I will double check, as well as improving this code.